apps/asap-server/test/handlers/research-output/index-handler.test.ts

```diff
@@ -20,36 +22,379 @@ describe('Research Output index handler', () => {
+        title: 'Original title',
+      });
+
+      await indexHandler(createEv);
+      expect(algoliaIndexMock.deleteObject).not.toHaveBeenCalled();
+      expect(algoliaIndexMock.saveObject).toHaveBeenCalledWith({
+        ...researchOutputResponse,
+        title: 'Original title',
+        objectID: researchOutputResponse.id,
+      });
+
+      researchOutputControllerMock.fetchById.mockResolvedValueOnce({
+        ...researchOutputResponse,
+        title: 'New title',
+      });
+
+      await indexHandler(updateEv);
+      expect(algoliaIndexMock.deleteObject).not.toHaveBeenCalled();
+      expect(algoliaIndexMock.saveObject).toHaveBeenCalledTimes(2);
+      expect(algoliaIndexMock.saveObject).toHaveBeenLastCalledWith({
+        ...researchOutputResponse,
+        title: 'New title',
+        objectID: researchOutputResponse.id,
+      });
+    });
+
+    it('receives on reverse order the events created and updated', async () => {
+      const roID = 'ro-1234';
+      const createEv = createEvent(roID);
+      const updateEv = updateEvent(roID);
+      const researchOutputResponse = {
+        ...getResearchOutputResponse(),
+        id: roID,
+      };
+
+      researchOutputControllerMock.fetchById.mockResolvedValueOnce({
+        ...researchOutputResponse,
+        title: 'Updated title',
+      });
+
+      await indexHandler(updateEv);
+      expect(algoliaIndexMock.deleteObject).not.toHaveBeenCalled();
+      expect(algoliaIndexMock.saveObject).toHaveBeenLastCalledWith({
+        ...researchOutputResponse,
+        title: 'Updated title',
+        objectID: researchOutputResponse.id,
+      });
+
+      researchOutputControllerMock.fetchById.mockResolvedValueOnce({
+        ...researchOutputResponse,
+        title: 'Old title',
+      });
+      await indexHandler(createEv);
+
+      expect(algoliaIndexMock.deleteObject).not.toHaveBeenCalled();
+      expect(algoliaIndexMock.saveObject).toHaveBeenLastCalledWith({
+        ...researchOutputResponse,
+        title: 'Old title',
+        objectID: researchOutputResponse.id,
+      });
+    });
+    it('receives on correct order the events created and unpublished', async () => {
+      const roID = 'ro-1234';
+      const createEv = createEvent(roID);
+      const unpublishedEv = unpublishedEvent(roID);
+
+      const researchOutputResponse = {
+        ...getResearchOutputResponse(),
+        id: roID,
+      };
+
+      researchOutputControllerMock.fetchById.mockResolvedValueOnce(
+        researchOutputResponse,
+      );
+
+      await indexHandler(createEv);
+      expect(algoliaIndexMock.deleteObject).not.toHaveBeenCalled();
+      expect(algoliaIndexMock.saveObject).toHaveBeenCalledWith({
+        ...researchOutputResponse,
+        objectID: researchOutputResponse.id,
+      });
+
+      await indexHandler(unpublishedEv);
+      expect(algoliaIndexMock.saveObject).toHaveBeenCalledTimes(1);
+      expect(algoliaIndexMock.deleteObject).toHaveBeenCalledWith(
```

Comment:
im actually surprised that this happens - shouldn't the delete only happen when `fetchById` throws? where does it say that it should throw in your test?